maintenance/dbconnect.py

```diff
@@ -12,11 +14,11 @@ class DBConnect(object):
     Class for connecting to a postgresql database using a single with statement
     """
     def __init__(self, dbhost='localhost', db='postgres', dbuser='postgres', dbpass='postgres',
-                 dbport=5432, autocommit=False, *args, **kwargs):
+                 dbport=5432, autocommit=False, cursor_factory=None, *args, **kwargs):
```

Comment:
here, you could define DBConnect.CURSOR_TYPE as a list or tuple.  ```DBConnect.CURSOR_TYPES = ['dict_cursor', 'default_cursor', None)``` 

then do a check on init to make sure the cursor_type was one of those vals or raise a TypeError if not.